grafana/grafana
# InfluxDB: Fix sending retention policy with InfluxQL queries

**What is this feature?**

From Original PR https://github.com/grafana/grafana/pull/62149
> In InfluxDB v2, due to technical limitations of the InfluxDB v1 compatibility layer, retention policies in a query (e.g. <retention policy>.<measurement_name>) aren't honored and must be specified in the URL query parameter rp to be applied. Grafana doesn't send this query parameter which results in all queries resolving to the default retention policy when querying InfluxDB v2 servers using InfluxQL.
> 
> This addresses the issue by sending the rp query parameter for queries that have specified a retention policy in the target given to runExploreQuery.
> 
> The outcomes are:
> 
>     InfluxQL queries executed against InfluxDB v2 databases will have the necessary retention policy information for queries like SHOW FIELD KEYS FROM measurement to function correctly.
>     InfluxQL queries executed against InfluxDB v1 databases will be unaffected, because this rp query parameter is unsupported there.
> 
> You can read more about the rentention policy mapping behavior of InfluxDB v2 in our documentation:
> 
>     https://docs.influxdata.com/influxdb/v2.6/reference/api/influxdb-1x/dbrp/#when-querying-data
>     https://docs.influxdata.com/influxdb/v2.6/reference/api/influxdb-1x/query/#query-a-non-default-retention-policy

**What did I add?**
I also changed the default retention policy value from `default` to `autogen`
Because: 
> InfluxDB automatically generated an RP named autogen and set it as the DEFAULT RP for the database.
>https://docs.influxdata.com/influxdb/v1.8/concepts/glossary/#retention-policy-rp

Without this we try to query with `default` retention policy and if there is no such policy, the query fails.
See: https://github.com/grafana/support-escalations/issues/5114

**Why do we need this feature?**

This ensures field keys only present in the default retention policy aren't displayed in the query builder for non-default retention policies.

**Who is this feature for?**

Anyone using InfluxQL to query against an InfluxDB v2 API (using v1 compatibility).

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/63672
https://github.com/grafana/support-escalations/issues/5114

### How to test
- Run Grafana
- Run dev env `make devenv sources=influxdb,influxdb1`
- Go to `Explore`
- Select `gdev-influxdb1-influxql`
- See in the `FROM` section, before `select measurement` the retention policies are loaded. 
- The default policy is the first one and selected
- Select a measurement and a field.
- Observe that you can do both
- Select `gdev-influxdb-influxql`
- Go through same steps
- In the network tab you would see your queries are being sent with policy. 
  - `bar` as policy
  - `SELECT mean("value") FROM "bar"."cpu" WHERE time >= now() - 1h and time <= now() GROUP BY time(1s) fill(null)`
